framework/behaviors/AttributeTypecastBehavior.php

```diff
@@ -0,0 +1,329 @@
+ *
+ * - after successful model validation
+ * - before model save (insert or update)
+ * - after model find (found by query or refreshed)
+ *
+ * You may disable automatic typecasting for particular case using fields [[typecastAfterValidate]],
+ * [[typecastBeforeSave]] and [[typecastAfterFind]].
+ *
+ * Note: you can manually trigger attribute typecasting anytime invoking [[typecastAttributes()]] method:
+ *
+ * ```php
+ * $model = new Item();
+ * $model->price = '38.5';
+ * $model->is_active = 1;
+ * $model->typecastAttributes();
+ * ```
+ *
+ * @property BaseActiveRecord $owner the owner of this behavior
+ *
+ * @author Paul Klimov <klimov.paul@gmail.com>
+ * @since 2.0.10
+ */
+class AttributeTypecastBehavior extends Behavior
+{
+    const TYPE_INTEGER = 'integer';
+    const TYPE_FLOAT = 'float';
+    const TYPE_BOOLEAN = 'boolean';
+    const TYPE_STRING = 'string';
+
+    /**
+     * @var array attribute typecast map in format: attributeName => type.
+     * Type can be set via PHP callable, which accept raw value as an argument and should return
+     * typecast result.
+     * For example:
+     *
+     * ```php
+     * [
+     *     'amount' => 'integer',
+     *     'price' => 'float',
+     *     'is_active' => 'boolean',
+     *     'date' => function ($value) {
+     *         return ($value instanceof \DateTime) ? $value->getTimestamp(): (int)$value;
+     *     },
+     * ]
+     * ```
+     *
+     * If not set, attribute type map will be composed automatically from the owner validation rules.
+     */
+    public $attributeTypes;
+    /**
+     * @var boolean whether to skip typecasting of `null` values.
+     * If enabled attribute value which equals to `null` will not be type-casted (e.g. `null` remains `null`),
+     * otherwise it will be converted according to the type configured at [[attributeTypes]].
+     */
+    public $skipOnNull = true;
+    /**
+     * @var boolean whether to perform typecasting after owner model validation.
+     * Note that typecasting will be performed only if validation was successful, e.g.
+     * owner model has no errors.
+     */
+    public $typecastAfterValidate = true;
+    /**
+     * @var boolean whether to perform typecasting before saving owner model (insert or update).
+     * This option may be disabled in order to achieve better performance.
+     * For example, in case of [[\yii\db\ActiveRecord]] usage, typecasting before save
+     * will grant no benefit an thus can be disabled.
+     */
+    public $typecastBeforeSave = true;
+    /**
+     * @var boolean whether to perform typecasting after retrieving owner model data from
+     * the database (after find or refresh).
+     * This option may be disabled in order to achieve better performance.
+     * For example, in case of [[\yii\db\ActiveRecord]] usage, typecasting after find
+     * will grant no benefit in most cases an thus can be disabled.
+     */
+    public $typecastAfterFind = true;
```

Comment:
Does this really make that much sense? This is done in the `yii\db\ColumnSchema::typecast` method.